RecoLocalMuon/CSCSegment/src/CSCSegAlgoRU.cc

```diff
@@ -0,0 +1,1038 @@
+  return (intercept + slope*layer); 
+} 
+
+void CSCSegAlgoRU::baseline(int n_seg_min){ 
+
+  int nhits      = proto_segment.size(); 
+  ChamberHitContainer::const_iterator iRH_worst; 
+
+  //initialise vectors for strip position and error within strip
+  SVector6 sp; 
+  SVector6 se;  
+
+  ChamberHitContainer buffer1; 
+  buffer1.clear(); 
+
+  unsigned int init_size = proto_segment.size(); 
+  while (buffer1.size()< init_size){ 
+    ChamberHitContainer::iterator min; 
+    int min_layer = 10; 
+    for(ChamberHitContainer::iterator k = proto_segment.begin(); k != proto_segment.end(); k++){       
+      const CSCRecHit2D* iRHk = *k;  
+      CSCDetId idRHk = (CSCDetId)(*iRHk).cscDetId(); 
+      int kLayer   = idRHk.layer(); 
+      if(kLayer < min_layer){ 
+	min_layer = kLayer; 
+	min = k; 
+      } 
+    } 
+    buffer1.push_back(*min); 
+    proto_segment.erase(min); 
+  }//while 
+  proto_segment.clear();  
+
+  for (ChamberHitContainer::const_iterator cand = buffer1.begin(); cand != buffer1.end(); cand++) { 
+    proto_segment.push_back(*cand); 
+  } 
+
+  for(ChamberHitContainer::const_iterator iRH = proto_segment.begin(); iRH != proto_segment.end(); iRH++){       
+    const CSCRecHit2D* iRHp = *iRH;
+    CSCDetId idRH = (CSCDetId)(*iRHp).cscDetId();
+    int kRing    = idRH.ring();
+    int kStation = idRH.station();
+    int kLayer = idRH.layer();
+    // Find the strip containing this hit                                                                                                                                             
+    int centerid     =  iRHp->nStrips()/2;
+    int centerStrip =  iRHp->channels(centerid);
+    float stpos = (*iRHp).positionWithinStrip();
+    se(kLayer-1) = (*iRHp).errorWithinStrip();
+    // Take into account half-strip staggering of layers (ME1/1 has no staggering)                                                                                                    
+    if (kStation == 1 && (kRing == 1 || kRing == 4)) sp(kLayer-1) = stpos + centerStrip;
+    else{
+      if (kLayer == 1 || kLayer == 3 || kLayer == 5) sp(kLayer-1) = stpos + centerStrip;
+      if (kLayer == 2 || kLayer == 4 || kLayer == 6) sp(kLayer-1) = stpos - 0.5 + centerStrip;
+    }
+  } 
+
+  float chi2_str; 
+  fitX(sp, se, chi2_str); 
+
+  //----------------------------------------------------- 
+  // Optimal point rejection method 
+  //----------------------------------------------------- 
+
+  float minSum = 1000; 
+  int i1b = 0; 
+  int i2b = 0;  
+  int iworst = -1;  
+  int bad_layer = -1; 
+  ChamberHitContainer::iterator rh_to_be_deleted_1; 
+  ChamberHitContainer::iterator rh_to_be_deleted_2; 
+  if ( nhits > n_seg_min && (chi2_str/(nhits-2)) > chi2_str_*chi2D_iadd){ 
+    for (ChamberHitContainer::iterator i1 = proto_segment.begin(); i1 != proto_segment.end();++i1) { 

```

Comment:
iterator changed to const_iterator in the entire baseline() function plus one identical change in replaceHit() function